src/main/java/tech/sourced/siva/RangeInputStream.java

```diff
@@ -0,0 +1,56 @@
+package tech.sourced.siva;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ *
+ */
+public class RangeInputStream extends InputStream {
+    private static final int READ_MASK = 0xFF;
+
+    private final InputStream parent;
+    private long remaining;
+
+    /**
+     * InputStream that will provide only the specified size of the provided InputStream
+     * from the position already provided on the parent stream.
+     *
+     * @param parent The parent InputStream where the data is coming from.
+     * @param size   Size of bytes to read from.
+     * @throws IOException If is not possible to skip the amount
+     *                     of bytes provided to skip or in any other standard InputStream case.
+     */
+    public RangeInputStream(
+            final InputStream parent,
+            final long size) throws IOException {
+        this.parent = parent;
+        remaining = size;
+    }
+
+    @Override
+    public final int read() throws IOException {
+        if (--remaining >= 0) {
```

Comment:
At a first sight, it looks weird, could be the decrement be placed before the `if statement` so then it just checks for the condition?

Reply:
It does not look weird to me, but I'm not neither for nor against changing it, so whatever you guys think its best.